### PR TITLE
Create poke card component

### PR DESCRIPTION
**Pull request by erecoba**
Creation of a poke card with a 3d animation when the cursor is over the card.

### CODE REVIEW

File: src/components/PokeCard.js [hunk 1]
```diff
@@ -0,0 +1,189 @@
+import React, { useState, useRef } from "react";
+import { mockPokemonData } from "../mock/pokeData";
```
Comment (wdonet) [inline src/components/PokeCard.js]: it is better if you load mock data out of the component, then pass the value as properties

File: src/components/PokeCard.js [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        rel="noreferrer"
+        style={styles.card.text}
+      >
+        {props.children}
+      </a>
+    </p>
+  );
+};
+
+const styles = {
```
Comment (wdonet) [inline src/components/PokeCard.js]: better to handle styling in a separate file

File: src/components/PokeCard.js [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+};
+
+const CardLinkVideo = (props) => {
+  return (
+    <p style={{ ...styles.card.text, ...props.dynamicStyle }}>
+      <a
+        href={props.href}
+        target="_blank"
+        rel="noreferrer"
+        style={styles.card.text}
```
Comment (wdonet) [inline src/components/PokeCard.js]: It is better to style components using classnames instead

File: src/components/PokeCard.js [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+    </div>
+  );
+}
+
+const Card = (props) => {
+  return (
+    <div
+      ref={props.parentRef}
+      style={{ ...styles.card.container, ...props.dynamicStyle }}
+      onMouseMove={props.onMouseMove}
```
Comment (wdonet) [inline src/components/PokeCard.js:112]: awesome! animation looks good 💯

**Review by wdonet — APPROVED** [review]
There's a warning in your code when running:
image

but Good job! 🤖  👍🏽

File: src/components/PokeCard.js (hunk 1 above)
Comment (erecoba) [reply src/components/PokeCard.js]: Yes, you're right, I already did this, thank you!

App.js
```javascript
import { mockPokemonData } from "./mock/pokeData";

export default function App() {
  return (
    <div className="App">
      <h1>Welcome to your second mini challenge!</h1>
      <PokeCard mockPokemonData={mockPokemonData} />
    </div>
  );
}
```

PokeCard.js
```javascript

function PokeCard({ mockPokemonData }) {// <---- here it comes
  const [cardStyle, setCardStyle] = useState(stylesNormalCard());
  const [spriteStyle, setSpriteStyle] = useState(styleNormalSprite());
  const [textStyle, setTextStyle] = useState(styleNormalText());
  const cardRef = useRef(null);

  ...
}
```

File: src/components/PokeCard.js (hunk 2 above)
Comment (erecoba) [reply src/components/PokeCard.js]: I remove that constant and I used a css file instead, like you suggested 😃, now it looks better.

PokeCard.js
```javascript
import React, { useState, useRef } from "react";
import "./PokeCard.css";

...

const Card = (props) => {
  return (
    <div
      ref={props.parentRef}
      className="PokeCard__card"// <--- A className here
      style={{ ...props.dynamicStyle }}
      onMouseMove={props.onMouseMove}
      onMouseLeave={props.onMouseLeave}
    >
      {props.children}
    </div>
  );
};

const CardTitle = (props) => {
  return (
    <h1 className="PokeCard__text" style={{ ...props.dynamicStyle }}>// <--- A className here
      {props.children}
    </h1>
  );
};

const CardSprites = (props) => {
  return (
    <div>
      {props.sprites.map((sprite, i) => {
        return (
          <img
            key={i}
            src={sprite.img}
            alt={props.alt}
            className="PokeCard__img"// <--- A className here
            style={{ ...props.dynamicStyle }}
          />
        );
      })}
    </div>
  );
};

const CardLinkVideo = (props) => {
  return (
    <p style={{ ...props.dynamicStyle }}>
      <a
        href={props.href}
        target="_blank"
        rel="noopener noreferrer"
        className="PokeCard__text"// <--- A className here
      >
        {props.children}
      </a>
    </p>
  );
};
```

File: src/components/PokeCard.js (hunk 3 above)
Comment (erecoba) [reply src/components/PokeCard.js]: It's done, now I looks better :D

File: src/components/PokeCard.js (hunk 4 above)
Comment (erecoba) [reply src/components/PokeCard.js:112]: thank you for your comments !!!!

**Pull request merged**